plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -457,46 +477,53 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
-    ).toContainer()
+    ).toProperty()
 
-    inline fun <reified D : Documentable> ClassDescriptor.additionalExtras(): PropertyContainer<D> = listOfNotNull(
+    fun ClassDescriptor.additionalExtras() = listOfNotNull(
         ExtraModifiers.DYNAMIC.takeIf { isDynamic() },
         ExtraModifiers.INLINE.takeIf { isInline },
         ExtraModifiers.EXTERNAL.takeIf { isExternal },
         ExtraModifiers.INNER.takeIf { isInner },
         ExtraModifiers.DATA.takeIf { isData },
         ExtraModifiers.OVERRIDE.takeIf { getSuperInterfaces().isNotEmpty() || getSuperClassNotAny() != null }
-    ).toContainer()
+    ).toProperty()
 
-    fun ValueParameterDescriptor.additionalExtras(): PropertyContainer<Parameter> =
+    fun ValueParameterDescriptor.additionalExtras() =
         listOfNotNull(
             ExtraModifiers.DYNAMIC.takeIf { isDynamic() },
             ExtraModifiers.NOINLINE.takeIf { isNoinline },
             ExtraModifiers.CROSSINLINE.takeIf { isCrossinline },
             ExtraModifiers.CONST.takeIf { isConst },
             ExtraModifiers.LATEINIT.takeIf { isLateInit },
             ExtraModifiers.VARARG.takeIf { isVararg }
-        ).toContainer()
+        ).toProperty()
 
-    fun TypeParameterDescriptor.additionalExtras(): PropertyContainer<TypeParameter> =
+    fun TypeParameterDescriptor.additionalExtras() =
         listOfNotNull(
             ExtraModifiers.DYNAMIC.takeIf { isDynamic() },
             ExtraModifiers.REIFIED.takeIf { isReified }
-        ).toContainer()
+        ).toProperty()
 
-    fun PropertyDescriptor.additionalExtras(): PropertyContainer<Property> = listOfNotNull(
+    fun PropertyDescriptor.additionalExtras() = listOfNotNull(
         ExtraModifiers.DYNAMIC.takeIf { isDynamic() },
         ExtraModifiers.CONST.takeIf { isConst },
         ExtraModifiers.LATEINIT.takeIf { isLateInit },
         ExtraModifiers.STATIC.takeIf { isJvmStaticInObjectOrClassOrInterface() },
         ExtraModifiers.EXTERNAL.takeIf { isExternal },
         ExtraModifiers.OVERRIDE.takeIf { DescriptorUtils.isOverride(this) }
-    ).toContainer()
+    ).toProperty()
+
+    fun List<ExtraModifiers>.toProperty() =
+        AdditionalModifiers(this)
+
+    fun DeclarationDescriptor.getAnnotations() = annotations.map { annotation ->
+        Annotations.Annotation(annotation?.let { it.annotationClass as DeclarationDescriptor }?.let { DRI.from(it) },
+            annotation.allValueArguments.map { (k, v) -> k.asString() to v.value.toString() }.toMap()
+        )
+    }.let { Annotations(it) }
 
-    inline fun <reified D : Documentable> List<ExtraModifiers>.toContainer(
-        container: PropertyContainer<D> = PropertyContainer.empty()
-    ): PropertyContainer<D> =
-        container + AdditionalModifiers(this)
+    inline fun <reified D : Documentable> toContainer(vararg extras: ExtraProperty<D>): PropertyContainer<D> =
+        extras.fold(PropertyContainer.empty(), { pc, p -> pc + p })
```

Comment:
Please create a method in the `PropertyContainer` like `addAll` or something similar

Reply:
Created.